File: cloud-collect-web/src/services/tagService.ts
import {Page, PageableParam, Tag} from "@/domain"
import http from "@/http"

const tagUrl = "/tag"

export async function create(tag: Tag) {
  await http.post(`${tagUrl}/create`, tag)
}

export async function modify(id: number, tag: Tag) {
  await http.put(`${tagUrl}/${id}`, tag)
}

export async function deleteById(id: number) {
  await http.delete(`${tagUrl}/${id}`)
}

export async function findById(id: number) {
  const response = await http.get<Tag>(`${tagUrl}/${id}`)
  return response.data
}

export async function findAll(pageableParam: PageableParam) {
  const params = {...pageableParam}
  const response = await http.get<Page<Tag>>(`${tagUrl}/findAll`, {params})
  return response.data
}

export async function findAllByNameContains(name: string, pageableParam: PageableParam) {
  const params = {name, ...pageableParam}
  const response = await http.get<Page<Tag>>(`${tagUrl}/findAllByNameContains`, {params})
  return response.data
}

//export async function findAllByUserId(userId: number, pageableParam: PageableParam) {
//  const params = {userId, ...pageableParam}
//  const response = await http.get<Page<Tag>>(`${tagUrl}/findAllByUserId`, {params})
//  return response.data
//}

//export async function findAllByNameContainsAndUserId(userId: number, name: string, pageableParam: PageableParam) {
//  const params = {userId, name, ...pageableParam}
//  const response = await http.get<Page<Tag>>(`${tagUrl}/findAllByNameContainsAndUserId`, {params})
//  return response.data
//}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/service/TagService.kt
package com.windea.demo.cloudcollect.core.service

import com.windea.demo.cloudcollect.core.domain.entity.*
import org.springframework.data.domain.*

interface TagService {
	/**创建自己的标签。*/
	fun create(tag: Tag, user: User)
	
	/**修改自己的标签。包括名字，概述*/
	fun modify(id: Long, tag: Tag)
	
	/**删除自己的标签。*/
	fun deleteById(id: Long)
	
	/**根据id得到某一标签。*/
	fun findById(id: Long): Tag
	
	/**得到所有标签。*/
	fun findAll(pageable: Pageable): Page<Tag>
	
	/**得到所有分类。*/
	fun findAllByNameContains(name: String, pageable: Pageable): Page<Tag>
	
	/**根据名字和用户id模糊查询所有标签。*/
	fun findAllByNameContainsAndUserId(name: String, userId: Long, pageable: Pageable): Page<Tag>
	
	/**根据用户id查询所有标签。*/
	fun findAllByUserId(userId: Long, pageable: Pageable): Page<Tag>
	
	/**检查某一标签是否已存在。*/
	fun existsByNameAndUser(name: String, user: User): Boolean
	
	
	/**得到该标签的所有收藏。*/
	fun getCollectPage(id: Long, pageable: Pageable): Page<Collect>
}
